bsip-0044.md

```diff
@@ -0,0 +1,173 @@
+
+If a satisfactory `preimage` is not provided to the HTLC before the stipulated `time lock` expires, the `depositor` may request the return of `escrow assets`. The HTLC will only evaluate transfer request from `depositor` and after `timeout threshold`, then return `escrow assets` to `depositor`.
+
+### **Condition Evaluators**
+
+The `preimage` can be thought of a secret key, that will eventually be shared with the `recipient`. This can be a word, a phrase, or even a random series of bytes. The `length` of the `preimage` must be specified within the HTLC at creation.
+
+Upon presentation of a `preimage`, the HTLC `condition evaluator` validates:
+
+1. That the `timeout threshold` has not yet occurred.
+
+2. That the length of the `preimage` matches the specified `preimage length`. 
+
+3. That the `hash of the preimage` matches the specified `preimage hash`. 
+
+If all evaluations succeed, the `escrow assets` are transferred to the `recipient`. If any evaluation fails, nothing happens; the HTLC remains ready to evaluate the next `preimage`.
+
+Upon presentation of a request from the `depositor` for return of `escrow assets`, the HTLC `condition evaluator` validates that the `timeout threshold` is now in the past. It then returns the `escrow assts` to `depositor`.
+
+### **Timing of Condition Evaluation**
+
+The `timeout threshold` of the contract is defined by `depositor` within the HTLC at creation. It can be any time in the future and should allow enough time for `recipient` to review the HTLC and provide the `preimage`. Further, it should not be set too far into the future to mitigate against an unresponsive `recipient` impacting `depositor`, as their `escrow assets` will be locked until `timeout threshold` expiry. The accuracy is based on when the `condition evaluator` runs, and should be considered accurate ± 15 seconds.
+
+### **Early Termination of an Escrow Contract**
+
+To protect the `recipient`, early termination of an HTLC is not allowed by any party. Placing a `timeout threshold` far into the future is valid. User protection from locking up funds for an extremely long period could be provided by the UI used to create the HTLC.
+
+### **Automatic Transfers Upon Expiry**
+
+Upon expiry of the `timeout threshold`, the `escrow assets` held within the HTLC will **not** automatically be returned to `depositor`. A separate transaction must be initiated by the `depositor` to request `escrow assets` be transferred to `depositor`. 
+
+Code _could_ be added to automate the return of funds. This could be part of block production or maintenance. However, this will cause extra load on those processes. Hence, it is felt that requiring action by the `depositor` is the best option.
```

Comment:
I disagree. *Not* returning the funds automatically will unnecessarily lock up memory resources for an indefinite period of time. Note that, typically, returning of funds will be requested sooner or later, so not returning them immediately doesn't save resources in the long run. Quite the opposite, since processing the return request transaction (including signature validation) will consume a lot more resources than the return operation itself.

Reply:
I agree that returning allocated memory is beneficial for validation nodes. I support automation when possible. It's unclear to me how/when the HTLC should be triggered for the automated return of assets to depositor. I feel it should not execute until LIB advances beyond contract expiry. Would appreciate some insights toward a spec on that. @jmjatlanta considered during maintenance.

Reply:
Thank you for you inputs. I was trying to prevent an extra step in maintenance, and was not looking at the bigger picture. I think the LIB argument is a good one, but would certainly like everyone else to weigh in.

Reply:
I don't think LIB makes sense here. If we release the fund immediately when got the input, the input and the release will be atomic. That said, if the chain reorganized and reverted the transaction with the input, the `release` action will be reverted automatically as well.